ci.sh

```diff
@@ -111,32 +126,38 @@ if [ "$PROVIDER_NAME" = "trusted-service" ]; then
     git submodule update --init
 fi
 
-echo "Build test"
-RUST_BACKTRACE=1 cargo build $FEATURES
+if [ -z "$RUN_COMMAND" ]; then
+    echo "Build test"
+    RUST_BACKTRACE=1 cargo build $FEATURES
 
-echo "Static checks"
-# On native target clippy or fmt might not be available.
-if rustup component list | grep -q fmt; then
-    cargo fmt --all -- --check
-fi
-if rustup component list | grep -q clippy; then
-    cargo clippy --all-targets $FEATURES -- -D clippy::all -D clippy::cargo
-fi
+    echo "Static checks"
+    # On native target clippy or fmt might not be available.
+    if rustup component list | grep -q fmt; then
+        cargo fmt --all -- --check
+    fi
+    if rustup component list | grep -q clippy; then
+        cargo clippy --all-targets $FEATURES -- -D clippy::all -D clippy::cargo
+    fi
 
-echo "Unit, doc and integration tests"
-RUST_BACKTRACE=1 cargo test $FEATURES
+    echo "Unit, doc and integration tests"
+    RUST_BACKTRACE=1 cargo test $FEATURES
 
-# Removing any mappings left over from integration tests
-rm -rf mappings/
+    # Removing any mappings left over from integration tests
+    rm -rf mappings/
 
-echo "Start Parsec for end-to-end tests"
-RUST_LOG=info RUST_BACKTRACE=1 cargo run $FEATURES -- --config $CONFIG_PATH &
-PARSEC_PID=$!
-# Sleep time needed to make sure Parsec is ready before launching the tests.
-sleep 5
+    echo "Start Parsec for end-to-end tests"
+    RUST_LOG=info RUST_BACKTRACE=1 cargo run $FEATURES -- --config $CONFIG_PATH &
+    # Sleep time needed to make sure Parsec is ready before launching the tests.
+    sleep 5
 
-# Check that Parsec successfully started and is running
-pgrep -f target/debug/parsec >/dev/null
+    # Check that Parsec successfully started and is running
+    pgrep -f target/debug/parsec >/dev/null
+else
+    # Start the service
+    eval $RUN_COMMAND
+
+    wait_for_service
+fi
 
 if [ "$PROVIDER_NAME" = "all" ]; then
```

Comment:
I understand that you modified this file as you want to re-use parts of it but I am thinking that if you actually look at what needs to be executed under tarpolin, it's not that much. I think it is mostly the normal tests and the stress tests. I don't think the persistence test would yield a lot more of code coverage.
My point being that, we could just put those same lines (normal tests and stress tests) directly in the `coverage.sh` script without touching this file at all.

Maybe later if we see that it is causing problems we can still go back but I don't see the lines duplicated changing much in the future

Reply:
Yep, can try - I wanted to run the persistence tests as well because those impact the startup methods for the providers which seem fairly important. It's annoying that you can't just execute parts of a bash script (or, I don't know how that would happen?). Maybe I can split `ci.sh` into multiple scripts, one for each type of test?

Reply:
(Oh, and the stress test doesn't get executed, it's skipped, because it doesn't add anything to coverage)

Reply:
> I wanted to run the persistence tests as well because those impact the startup methods for the providers which seem fairly important

ah I see. I guess for now you can just copy and paste the lines, in lack for a better way. We had an issue anyway to have a better testing infrastructure, with maybe more stuff in Rust